app/models/sam_account_reckoner.rb

```diff
@@ -0,0 +1,20 @@
+class SamAccountReckoner < Struct.new(:user)
```

Comment:
Namespace might be good, but this is not so much a decorator. Another alternative that I have used and Sandi Metz has talked about using in putting your AR records in a namespace. That sets up the expectation that those will be the minority of work, and that you name your other models according to what they do and are about. I think it is possible that the decorators might break up into view model, specific to a particular view representation, and areas of responsibility around a data model like this one.